py/dml/ctree.py

```diff
@@ -475,16 +475,26 @@ def toc(self):
 mkInline = Inline
 
 class InlinedMethod(Statement):
-    '''Wraps the body of an inlined method, to protect it from analysis'''
+    '''Wraps the body of an inlined method, both to protect it from analysis
+       and to ensure linemarks are created for it'''
     @auto_init
-    def __init__(self, site, method, body): pass
+    def __init__(self, site, method, pre, body, post): pass
     def toc(self):
-        self.body.toc()
+        out('{\n', postindent = 1)
+        self.toc_inline()
+        out('}\n', preindent = -1)
     def toc_inline(self):
-        self.body.toc_inline()
+        for stmt in self.pre:
+            stmt.toc()
+        with allow_linemarks():
+            for stmt in self.body:
+                stmt.toc()
+        for stmt in self.post:
+            stmt.toc()
     def control_flow(self):
         return ControlFlow(fallthrough=True, throw=self.method.throws)
 
+# TODO: should this rewrite `body=[Compound(statements)]` to `body=statements`?
 mkInlinedMethod = InlinedMethod
```

Comment:
Yeah it should, as long `statements` as it doesn't have any declaration. Previously `InlinedMethod` was just a wrapper around a `mkCompound(_; pre + body + post)`, and `mkCompound` collapses any `Compound` of the list is passed if it has no declaration.